cli/cmd/encore/auth/auth.go

```diff
@@ -30,14 +27,14 @@ func init() {
 
 		DisableFlagsInUseLine: true,
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := DoLogin(); err != nil {
+			if err := DoLogin(DeviceAuth); err != nil {
```

Comment:
why deviceauth here?

Reply:
Because it's more general. I kept the original flow around for the `encore app create` flow because it doesn't require a bunch of instructions that are unrelated to the task at hand (creating an app). If you explicitly run `encore auth login` that's not as much of a problem: you're explicitly invoking the login flow.

If the auto-detection is wrong we don't currently have a way of selecting a specific flow, so it seemed easier to default this flow to always use the device auth flow.

Reply:
fair. I guess an optional flag could be an option too.

Reply:
Yeah, it's just not very discoverable and if you have an issue it's not super-clear what to do. I figured it's easier to just always use the new flow in the `encore auth login` case.